src/crossEnvCrypto.ts

```diff
@@ -0,0 +1,57 @@
+import { HashFunction } from "./parameters";
+
+interface CompatibleCrypto {
+  hashFunctions: { [key: string]: HashFunction };
+  randomBytes: (array: Uint8Array) => Uint8Array;
+}
+
+export let crossEnvCrypto: CompatibleCrypto;
+
+try {
+  const webcrypto =
+    (typeof window !== "undefined" && window.crypto) ||
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require("crypto").webcrypto; // Node v15+ has webcrypto built in, lets use that if we can
+
+  if (webcrypto) {
+    if (!webcrypto.subtle) {
+      throw new Error(
+        "Crypto.Subtle is undefined. Make sure you are using HTTPS in a compatible browser.",
+      );
+    }
+    const digestFunctionToHashFunction =
+      (algorithm: AlgorithmIdentifier) => (data: ArrayBuffer) =>
+        webcrypto.subtle.digest(algorithm, data);
+    crossEnvCrypto = {
+      randomBytes: webcrypto.getRandomValues,
```

Comment:
This throws `IllegalInvocation`, it needs to be `webcrypto.getRandomValues.bind(webcrypto)`, or use an arrow fn, whichever you prefer.